misc/acrn-config/board_config/vbar_base_h.py

```diff
@@ -128,6 +138,66 @@ def find_next_bar(bar_val, bar_list):
     return idx
 
 
+def write_vuart_vbar(mmiolist_per_vm, sos_mmio_range, config):
+    # get legacy vuart information
+    vuart0_setting = common.get_vuart_info_id(common.SCENARIO_INFO_FILE, 0)
+    vuart1_setting = common.get_vuart_info_id(common.SCENARIO_INFO_FILE, 1)
+    # get pci vuart information
+    vuarts = common.get_vuart_info(common.SCENARIO_INFO_FILE)
+    for vm_id in vuarts.keys():
+        vm_type = common.VM_TYPES[vm_id]
+        if scenario_cfg_lib.VM_DB[vm_type]['load_type'] == "POST_LAUNCHED_VM":
+            continue
+        for vuart_id in vuarts[vm_id].keys():
+            if vuarts[vm_id][vuart_id]['base'] == "INVALID_PCI_BASE":
+                continue
+            # Skip pci vuart 0 if the legacy vuart 0 is enabled
+            if vuart_id == 0 and vm_id in vuart0_setting \
+                 and vuart0_setting[vm_id]['base'] != "INVALID_COM_BASE":
+                continue
+            # Skip pci vuart 1 if the legacy vuart 1 is enabled
+            if vuart_id == 1 and vm_id in vuart1_setting \
+                 and vuart1_setting[vm_id]['base'] != "INVALID_COM_BASE":
+                continue
+            free_bar0 = []
+            free_bar1 = []
+            # vuart decice requires 2 bars
+            if scenario_cfg_lib.VM_DB[vm_type]['load_type'] == "SOS_VM":
+                free_bar0 = get_free_mmio(sos_mmio_range, mmiolist_per_vm[vm_id], \
+                             PCI_VUART_VBAR0_SIZE + PCI_VUART_VBAR0_ALIGNMENT)
+                free_bar0_start_addr = common.round_up(free_bar0.start, PCI_VUART_VBAR0_ALIGNMENT)
+                free_bar0_end_addr = free_bar0_start_addr + PCI_VUART_VBAR0_SIZE - 1
+                free_bar0 = MmioWindow(free_bar0_start_addr, free_bar0_end_addr)
+                mmiolist_per_vm[vm_id].append(free_bar0)
+                mmiolist_per_vm[vm_id].sort()
+                free_bar1 = get_free_mmio(sos_mmio_range, mmiolist_per_vm[vm_id], \
+                             PCI_VUART_VBAR1_SIZE + PCI_VUART_VBAR1_ALIGNMENT)
+                free_bar1_start_addr = common.round_up(free_bar1.start, PCI_VUART_VBAR1_ALIGNMENT)
+                free_bar1_end_addr = free_bar0_start_addr + PCI_VUART_VBAR1_SIZE - 1
```

Comment:
I guess this line should be:
free_bar1_end_addr = free_bar1_start_addr + PCI_VUART_VBAR1_SIZE - 1

Reply:
fixed